src/main/java/org/mitre/synthea/world/agents/Person.java

```diff
@@ -611,42 +615,24 @@ public void setCurrentEncounter(Module module, Encounter encounter) {
 
   // Providers API -----------------------------------------------------------
   public static final String CURRENTPROVIDER = "currentProvider";
-  public static final String PREFERREDYPROVIDER = "preferredProvider";
 
   /**
    * Get the preferred provider for the specified encounter type. If none is set the
    * provider at the specified time as the preferred provider for this encounter type.
    */
-  public Provider getProvider(EncounterType type, long time) {
-    String key = PREFERREDYPROVIDER + type;
-    if (!attributes.containsKey(key)) {
-      setProvider(type, time);
+  public Provider getProvider(EncounterType type, String speciality, long time) {
+    if (!preferredProviders.doesRelationshipExist(type, speciality)) {
+      setProvider(type, speciality, time);
     }
```

Comment:
Thoughts on having both `getProvider(EncounterType, long)` and `getProvider(EncounterType, String, long)`?

Where the first is a one-liner that passes a `null` to the String argument of the other function, rather than have `null` arguments all throughout the code?

Seems weird to me as-is.

Reply:
My $0.02 is that I like the `null`s. The reality is that if the PR is accepted, Synthea is in a little bit of an odd state. In the past, there really wasn't specialty, so everything sort of being general practice was OK. Now, there are likely places where we don't want the default behavior. Having the `null` is a place holder that a future effort can come along and fill in.